.github/workflows/auto-approve-cicd-release-pr.yml

```diff
@@ -0,0 +1,17 @@
+name: Auto-approve temporal-cicd[bot] PRs on cloud release branches
+on:
+  pull_request:
+    branches:
+      - cloud/**
+permissions:
+  contents: write
+  pull-requests: write
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'temporal-cicd[bot]' }}
```

Comment:
I'm not really concerned, but just curious, how is this authenticated?

Reply:
I guess I mean, is it possible for someone to fake that "actor" name by setting their github username or something? Or creating a PR and somehow tricking the bot into taking an action on it? I found this article which describes something like that: https://www.synacktiv.com/en/publications/github-actions-exploitation-dependabot
It looks like that probably only works with dependabot, but it suggests (at least to me) that this whole thing is pretty subtle.

Is it possible to take another approach and have the bot run `gh pr review --approve` or use the API directly, similar to how it creates the PR in the first place, instead of using GHA?